pkg/backend/update.go

```diff
@@ -86,37 +86,12 @@ func (c *UpdateCommand) Exec(in io.Reader, out io.Writer) error {
 		return err
 	}
 
-	// Copy existing values from GET to UpdateBackendInput strcuture
 	input := &fastly.UpdateBackendInput{
-		ServiceID:           b.ServiceID,
-		ServiceVersion:      b.ServiceVersion,
-		Name:                b.Name,
-		NewName:             fastly.String(b.Name),
-		Address:             fastly.String(b.Address),
-		Port:                fastly.Uint(b.Port),
-		OverrideHost:        fastly.String(b.OverrideHost),
-		ConnectTimeout:      fastly.Uint(b.ConnectTimeout),
-		MaxConn:             fastly.Uint(b.MaxConn),
-		FirstByteTimeout:    fastly.Uint(b.FirstByteTimeout),
-		BetweenBytesTimeout: fastly.Uint(b.BetweenBytesTimeout),
-		AutoLoadbalance:     fastly.CBool(b.AutoLoadbalance),
-		Weight:              fastly.Uint(b.Weight),
-		RequestCondition:    fastly.String(b.RequestCondition),
-		HealthCheck:         fastly.String(b.HealthCheck),
-		Shield:              fastly.String(b.Shield),
-		UseSSL:              fastly.CBool(b.UseSSL),
-		SSLCheckCert:        fastly.CBool(b.SSLCheckCert),
-		SSLCACert:           fastly.String(b.SSLCACert),
-		SSLClientCert:       fastly.String(b.SSLClientCert),
-		SSLClientKey:        fastly.String(b.SSLClientKey),
-		SSLCertHostname:     fastly.String(b.SSLCertHostname),
-		SSLSNIHostname:      fastly.String(b.SSLSNIHostname),
-		MinTLSVersion:       fastly.String(b.MinTLSVersion),
-		MaxTLSVersion:       fastly.String(b.MaxTLSVersion),
-		SSLCiphers:          b.SSLCiphers,
-	}
-
-	// Set values to existing ones to prevent accidental overwrite if empty.
+		ServiceID:      b.ServiceID,
+		ServiceVersion: b.ServiceVersion,
+		Name:           b.Name,
+	}
```

Comment:
@kailan just realised that my previous changes weren't necessary. All I had to do in fact was delete this chunk of code where it sets values into the update struct for otherwise optional fields.

The reason this works is because in go-fastly we have all the optional fields in an update struct as 'pointers' which means if the fields are never set here in the CLI where it makes the go-fastly call, then those fields won't get passed through to the upstream API.

The only time the fields get set is later on in this file when we use `WasSet` to indicate an _actual_ change in the field was provided by the user.

Reply:
I'm going to open a separate PR to ensure this pattern (which is what we use in terraform but never went back and implemented in the CLI) is applied throughout the CLI (I'm not going to do it here in this PR).